tests/components/shelly/conftest.py

```diff
@@ -155,9 +163,89 @@ async def coap_wrapper(hass):
     return wrapper
 
 
+@pytest.fixture
+async def rest_wrapper(hass):
+    """Setups a rest wrapper with mocked device."""
+    await async_setup_component(hass, "shelly", {})
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={"sleep_period": 0, "model": "SHSW-25", "host": "1.2.3.4"},
+        unique_id="12345678",
+    )
+    config_entry.add_to_hass(hass)
+
+    device = Mock(
+        blocks=MOCK_BLOCKS,
+        settings=MOCK_SETTINGS,
+        shelly=MOCK_SHELLY,
+        status=MOCK_STATUS_COAP,
+        firmware_version="some fw string",
+        update=AsyncMock(),
+        trigger_ota_update=AsyncMock(),
+        trigger_reboot=AsyncMock(),
+        initialized=True,
+    )
+
+    hass.data[DOMAIN] = {DATA_CONFIG_ENTRY: {}}
+    hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id] = {}
+    wrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id][
+        REST
+    ] = ShellyDeviceRestWrapper(hass, device, config_entry)
+
+    block_wrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id][
+        BLOCK
+    ] = BlockDeviceWrapper(hass, config_entry, device)
+    block_wrapper.async_setup()
+
+    return wrapper
+
+
+@pytest.fixture
+async def rest_wrapper_none(hass):
```

Comment:
Do not duplicated the wrappers just to modify the data, instead use monkepatch:
https://github.com/home-assistant/core/blob/9e8b542319117f8884ba717a3632b82c9dc7820f/tests/components/shelly/test_cover.py#L183

It would also be better to have a single test that checks for update status, patch the data and check again that update status is changed

Reply:
thank you, I am learning each time.
I have greathly simplified and cleaned up the tests.